### PR TITLE
Dev books

### PR DESCRIPTION
**Pull request by peny6jo**

### CODE REVIEW

File: books/api/bills.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 from dataclasses import dataclass
+from distutils.log import error
```
Comment (jesfel) [inline books/api/bills.py]: remove this line

File: books/api/credits_notes.py [hunk 2]
```diff
@@ -173,7 +173,7 @@ def update_billing_address(self,creditnote_id:str, data:dict):
                 "fax": "+1-925-924-9600"
             }
         """
-        return self.base.api._standard_call(f'{self.module}/{creditnote_id}/address/billing','put')
+        return self.base.api._standard_call(f'{self.module}/{creditnote_id}/address/billing','put', data=data)
```
Comment (jesfel) [inline books/api/credits_notes.py]: data.json? or it is not necessary?

File: books/api/invoice.py [hunk 3]
```diff
@@ -67,7 +67,9 @@ def get_all(self, page:int=1, invoice_number:str= None, reference_number:str= No
         Returns:
             Response
         """
-        return self.base.api._standard_call(f'{self.module}', 'get', page=page)
+        return self.base.api._standard_call(f'{self.module}', 'get', page=page, invoice_number=invoice_number, reference_number=reference_number, customer_name=customer_name, total=total,\
```
Comment (jesfel) [inline books/api/invoice.py:70]: We should test this method, what happen if we don't pass any optional parameter?

File: books/api/taxes.py [hunk 4]
```diff
@@ -36,7 +36,7 @@ def update (self, tax_id:str, data:TaxData):
             data (TaxData)
             tax_id(str)
         """
-        return self.base.api._standard_call(f'{self.module}/taxes{tax_id}', 'put', data=data)
+        return self.base.api._standard_call(f'{self.module}/taxes{tax_id}', 'put', data=str(data))
```
Comment (jesfel) [inline books/api/taxes.py]: is .json not needed?

File: books/api/credits_notes.py (hunk 2 above)
Comment (peny6jo) [reply books/api/credits_notes.py]: Im not sure, maybe like data is dict is not neccesary

Comment (peny6jo) [reply books/api/credits_notes.py]: maybe just need convert to str

**Pull request merged**